pkg/providers/instancetype/instancetype.go

```diff
@@ -80,16 +84,18 @@ type DefaultProvider struct {
 	instanceTypesOfferingsSeqNum uint64
 }
 
-func NewDefaultProvider(region, clusterCNI string, ecsClient *ecsclient.Client,
+func NewDefaultProvider(region, clusterCNI string, ecsClient *ecsclient.Client, vpcapi *vpcclient.Client,
 	instanceTypesCache *cache.Cache, unavailableOfferingsCache *kcache.UnavailableOfferings,
 	pricingProvider pricing.Provider, vSwitchProvider vswitch.Provider) *DefaultProvider {
 	return &DefaultProvider{
 		ecsClient:              ecsClient,
+		vpcapi:                 vpcapi,
 		region:                 region,
 		clusterCNI:             clusterCNI,
 		vSwitchProvider:        vSwitchProvider,
 		pricingProvider:        pricingProvider,
 		instanceTypesInfo:      []*ecsclient.DescribeInstanceTypesResponseBodyInstanceTypesInstanceType{},
+		vSwitchInfo:            []*vpcclient.DescribeVSwitchesResponseBodyVSwitchesVSwitch{},
```

Comment:
Wouldn’t it be better to get the vSwitch information from the VSwitchProvider? Rather than maintaining two copies of the information.